pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +119,56 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	trustedCAName := "user-ca-bundle"
 	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
 		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
 		"-n", "openshift-config", "--type", "merge"}
 
 	if err := WaitForOpenshiftResource(ocConfig, "proxy"); err != nil {
 		return err
 	}
+	if proxy.ProxyCACert != "" {
+		logging.Debug("Adding proxy CA cert to cluster")
+		if err := addProxyCACertToCluster(sshRunner, ocConfig, proxy, trustedCAName); err != nil {
+			return err
+		}
+		cmdArgs = []string{"patch", "proxy", "cluster", "-p",
+			fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s", "trustedCA":{"name":"%s"}}}'`,
+				proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString(), trustedCAName),
+			"-n", "openshift-config", "--type", "merge"}
+	}
 	if _, stderr, err := ocConfig.RunOcCommand(cmdArgs...); err != nil {
 		return fmt.Errorf("Failed to add proxy details %v: %s", err, stderr)
 	}
 	return nil
 }
 
+func addProxyCACertToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig, trustedCAName string) error {
+	proxyCABundleTemplate := `{
+  "apiVersion": "v1",
+  "data": {
+    "ca-bundle.crt": "%s"
+  },
+  "kind": "ConfigMap",
+  "metadata": {
+    "name": "%s",
+    "namespace": "openshift-config"
+  }
+}
+`
+	p := fmt.Sprintf(proxyCABundleTemplate, strings.ReplaceAll(proxy.ProxyCACert, "\n", `\n`), trustedCAName)
+	err := sshRunner.SetTextContentAsRoot(fmt.Sprintf("/tmp/%s.json", trustedCAName), p, 0644)
+	if err != nil {
+		return err
+	}
+	cmdArgs := []string{"apply", "-f", fmt.Sprintf("/tmp/%s.json", trustedCAName)}
```

Comment:
`tmpFilename := fmt.Sprintf("/tmp/%s.json", trustedCAName)` and then use that. Maybe go with `filepath.Join()` and `filepath.ToSlash()`?

Reply:
The name of the tmp file is still generated twice. I would hardcode `user-ca-bundle` in `addProxyCACertToCluster`, no need for it to be an argument, as OpenShift expects it to be named this way if I'm not mistaken, so if you want to add a CA cert to the cluster, you have to use the correct `user-ca-bundle` name.

Reply:
@cfergeau https://docs.openshift.com/container-platform/4.5/networking/enable-cluster-wide-proxy.html#nw-proxy-configure-object_config-cluster-wide-proxy it says `user-ca-bundle` but name can be anything because it's the configmap name which we need to match here, about twice generating the `/tmp` file I can move it to top block of function.